nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/FlowResource.java

```diff
@@ -1358,6 +1359,41 @@ public Response getReportingTaskTypes(
         return generateOkResponse(entity).build();
     }
 
+    @GET
+    @Consumes(MediaType.WILDCARD)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("runtime-manifest")
+    @ApiOperation(
+            value = "Retrieves the runtime manifest for this NiFi instance.",
+            notes = NON_GUARANTEED_ENDPOINT,
+            response = RuntimeManifest.class,
+            authorizations = {
+                    @Authorization(value = "Read - /flow")
+            }
+    )
+    @ApiResponses(
+            value = {
+                    @ApiResponse(code = 400, message = "NiFi was unable to complete the request because it was invalid. The request should not be retried without modification."),
+                    @ApiResponse(code = 401, message = "Client could not be authenticated."),
+                    @ApiResponse(code = 403, message = "Client is not authorized to make this request."),
+                    @ApiResponse(code = 409, message = "The request was valid but NiFi was not in the appropriate state to process it. Retrying the same request later may be successful.")
+            }
+    )
+    public Response getRuntimeManifest() throws InterruptedException {
+
+        authorizeFlow();
+
+        if (isReplicateRequest()) {
+            return replicate(HttpMethod.GET);
+        }
```

Comment:
I debated about this, but I based it off the end-points that are currently used to retrieve available types for processors/controller-services/reporting-tasks, and since those don't replicate, I figured it was ok to leave this one not replicating until we come up with a good reason that it needs to.

Reply:
Thanks for the reply. Since it is a `GET` request, it seems reasonable that the request should not be replicated.  To clarify, should this conditional block be removed since the request will not be replicated?

Reply:
I stand corrected about the existing "types" end-points, they do actually perform replication and response merging. The idea is to provide a consistent response to a client with the intersection of the types in case one node has some NARs that another node doesn't have.

I pushed another commit that adds a response merger for the runtime-manifest. Right now the logic is to just perform the intersection of the bundles based on their unique group-artifact-version. To test this you can run a cluster and remove a NAR from one of the nodes and it should no longer show up in the manifest response.